### ISSUE TITLE
voms.mapfile does not evaluate environment variable

### ISSUE THREAD

**Issue by juztas** [xrootd/xrootd#2200]
Looks like xrootd does not evaluate and use correct env variable for voms.mapfile. 
Environment vars:
```
VOMS_MAP_LOCATION=/etc/grid-security/voms-mapfile
GRID_MAP_LOCATION=/etc/grid-security/grid-mapfile
```
and xrootd config:
```
set vomsmaplocation = $VOMS_MAP_LOCATION
set gridmaplocation = $GRID_MAP_LOCATION
voms.mapfile $vomsmaplocation
sec.protocol /usr/lib64 gsi -certdir:/etc/grid-security/certificates -cert:$(xrdcert) -key:$(xrdkey) -crl:1 -gmapopt:trymap -gridmap:$gridmaplocation -vomsfun:default
```
Once xrootd starts - here is from the log:
```
240220 19:09:09 1532 XrdVomsConfig: Will initialize VOMS mapfile $vomsmaplocation
240220 19:09:09 1532 XrdVomsXrdVomsMapfile: Unable to Error checking the mapfile $vomsmaplocation; no such file or directory
240220 19:09:09 1532 secgsi_LoadVOMSFun: problems executing 'XrdSecgsiVOMSInit()' (rc: -1)
240220 19:09:09 1532 secgsi_Init: Secgsi: ErrError: VOMS plug-in loading failed: libXrdVoms.so
Secgsi: ErrError: VOMS plug-in loading failed: libXrdVoms.so
Config Failed to load gsi authentication protocol!
=====> sec.protocol /usr/lib64 gsi -certdir:/etc/grid-security/certificates -cert:/etc/grid-security/xrootd/xrootdcert.pem -key:/etc/grid-security/xrootd/xrootdkey.pem -crl:1 -gmapopt:trymap -gridmap:/etc/grid-security/grid-mapfile -vomsfun:default
```
So it does not evaluate and use correct env variable of $vomsmaplocation, while ok for gridmapfile. Just to make sure it loads env variable, I modified sec.protocol line to load vomsmaplocation variable
```
OLD: sec.protocol /usr/lib64 gsi -certdir:/etc/grid-security/certificates -cert:$(xrdcert) -key:$(xrdkey) -crl:1 -gmapopt:trymap -gridmap:$gridmaplocation -vomsfun:default
NEW: sec.protocol /usr/lib64 gsi -certdir:/etc/grid-security/certificates -cert:$(xrdcert) -key:$(xrdkey) -crl:1 -gmapopt:trymap -gridmap:$vomsmaplocation -vomsfun:default
```
And it does use correct var for sec.protocol, so env variable is evaluated to correct path:
```
240220 19:11:21 1582 XrdVomsXrdVomsMapfile: Unable to Error checking the mapfile $vomsmaplocation; no such file or directory
240220 19:11:21 1582 secgsi_LoadVOMSFun: problems executing 'XrdSecgsiVOMSInit()' (rc: -1)
240220 19:11:21 1582 secgsi_Init: Secgsi: ErrError: VOMS plug-in loading failed: libXrdVoms.so
Secgsi: ErrError: VOMS plug-in loading failed: libXrdVoms.so
Config Failed to load gsi authentication protocol!
=====> sec.protocol /usr/lib64 gsi -certdir:/etc/grid-security/certificates -cert:/etc/grid-security/xrootd/xrootdcert.pem -key:/etc/grid-security/xrootd/xrootdkey.pem -crl:1 -gmapopt:trymap -gridmap:/etc/grid-security/voms-mapfile -vomsfun:default
```

**Comment by amadio** [comment]
@juztas Is the above a workable solution for your issue? We'd like to know if any action is still needed to resolve this ticket.

**Comment by juztas** [comment]
Mh.. The new way I will need to test it in nonproduction and see which params work and or should be done for cms. 

Does it mean `voms.mapfile` should not be used anymore?
I see an issue mainly here:
env variable: `VOMS_MAP_LOCATION=/etc/grid-security/voms-mapfile`
Inside xroot `voms.mapfile $vomsmaplocation`
It fails with:
```
240220 19:11:21 1582 XrdVomsXrdVomsMapfile: Unable to Error checking the mapfile $vomsmaplocation; no such file or directory
```
but if I do inside xrootd config like this (without $var and exact path to file):
```
voms.mapfile /etc/grid-security/voms-mapfile
```
It works. Looks like it does not evaluate $VAR which points to correct location file.

**Comment by abh3** [comment]
This will be fixed in the upcoming 5.7.0 via commit 5753199

**Comment by amadio** [comment]
Ok, I'm closing this then. Thanks, @abh3!
